src/goto-analyzer/static_simplifier.cpp

```diff
@@ -220,6 +221,10 @@ bool static_simplifier(
              << ", function calls: " << unmodified.function_calls
              << messaget::eom;
 
+  // restore return types before writing the binary
+  restore_returns(goto_model);
+  goto_model.goto_functions.update();
```

Comment:
Sure.  I know I keep saying this BUT we (in the broadest sense) do need to work out what the valid forms of goto-program are at various points in the tool.  This reads to me a little bit like "returns must always be present when on disk".

Reply:
Fair.